auth0/resource_auth0_action_test.go

```diff
@@ -179,3 +198,74 @@ resource auth0_action my_action {
 	}
 }
 `
+
+func TestCheckForUntrackedActionSecrets(t *testing.T) {
+	var testCases = []struct {
+		name                  string
+		givenOldSecretsConfig []interface{}
+		givenNewSecretsConfig []interface{}
+		givenActionSecrets    []*management.ActionSecret
+		expectedDiagnostics   diag.Diagnostics
+	}{
+		{
+			name:                  "action has no secrets",
+			givenOldSecretsConfig: []interface{}{},
+			givenNewSecretsConfig: []interface{}{},
+			givenActionSecrets:    []*management.ActionSecret{},
+			expectedDiagnostics:   diag.Diagnostics(nil),
+		},
+		{
+			name:                  "action has no untracked secrets",
+			givenOldSecretsConfig: []interface{}{},
+			givenNewSecretsConfig: []interface{}{
+				map[string]interface{}{
+					"name": "secretName",
+				},
+			},
+			givenActionSecrets: []*management.ActionSecret{
+				{
+					Name: auth0.String("secretName"),
+				},
+			},
+			expectedDiagnostics: diag.Diagnostics(nil),
+		},
+		{
+			name:                  "action has untracked secrets",
+			givenOldSecretsConfig: []interface{}{},
+			givenNewSecretsConfig: []interface{}{
+				map[string]interface{}{
+					"name": "secretName",
+				},
+			},
+			givenActionSecrets: []*management.ActionSecret{
+				{
+					Name: auth0.String("secretName"),
+				},
+				{
+					Name: auth0.String("anotherSecretName"),
+				},
+			},
+			expectedDiagnostics: diag.Diagnostics{
+				{
+					Severity: diag.Error,
+					Summary:  "Unmanaged Action Secret",
+					Detail: "Found unmanaged action secret with key: anotherSecretName. " +
+						"Add this secret to your configuration so it does not get wiped.",
+					AttributePath: cty.Path{cty.GetAttrStep{Name: "secrets"}},
+				},
+			},
+		},
+	}
+
```

Comment:
Perhaps an additional test case would be if intentionally attempting to delete an action secret? Although, if you consolidate the old and new secret slices, you may not be able to express that as a test case. Mostly thinking out loud here!